mindsdb/integrations/handlers/anyscale_endpoints_handler/anyscale_endpoints_handler.py

```diff
@@ -105,19 +106,46 @@ def _prepare_ft_jsonl(self, df, temp_storage_path, temp_filename, _, test_size=0
             df: has exactly two columns, `role` and `content`. Rows contain >= 1 chats in long (stacked) format.
             For more details, check `FineTuning -> Data Format` in the Anyscale API reference.
         """
+        def _is_valid(chat):
+            """ Check if chat is valid according to Anyscale criteria."""
+            roles = [m['role'] for m in chat]
+            transitions = {None: ['system', 'user'], 'system': ['user'], 'user': ['assistant'], 'assistant': ['user']}
+
+            # check base condition
+            if not ('user' in roles and 'assistant' in roles):
+                return False
+
+            # check order is valid
+            state = None
+            for role in roles:
+                if role not in transitions[state]:
+                    return False
+                else:
+                    state = role
+
+            # chat is valid, return
+            return True
+
         # 1. aggregate each chat sequence into one row
         chats = []
         chat = []
         for i, row in df.iterrows():
             if row['role'] == 'system' and len(chat) > 0:
-                chats.append({'messages': chat})
+                if _is_valid(chat):
+                    chats.append({'messages': chat})
                 chat = []
             event = {'role': row['role'], 'content': row['content']}
             chat.append(event)
-        chats.append({'messages': chat})
+
+        if _is_valid(chat):
+            chats.append({'messages': chat})
+
         series = pd.Series(chats)
-        train = series.iloc[:int(len(series) * (1 - test_size))]
-        val = series.iloc[-int(len(series) * test_size) - 1:]
+        if len(series) < 20 * 2:
```

Comment:
`20` for train is arbitrary, but the minimum size of `20` for validation is not.